### PR TITLE
api endpoints integration

### PR DESCRIPTION
**Pull request by WuFan9627**
integrate with login, fix the get_activity endpoint issue, add user login/register test

### CODE REVIEW

**Comment by tschelbs18** [comment]
@WuFan9627 can you please resolve merge conflicts first?

**Comment by WuFan9627** [comment]
> @WuFan9627 can you please resolve merge conflicts first?

File: src/habit_server/db_manager.py [hunk 1]
```diff
@@ -100,14 +94,17 @@ def delete_habit(self, habit):
         if len(habits) == 0:
             return Result.Err("Habit does not exist, cannot delete")
         else:
+            habit = self._session.query(UserHabit).filter_by(
+            username=habit.username,
+            habitname=habit.habitname
+            ).first()
             self._session.delete(habit)
```
Comment (jrdbuch) [inline src/habit_server/db_manager.py:101]: just do habits[0], no need to do the query again

File: test/test_api.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import uuid
+import pytest
+from flask import Flask, session, json
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import create_engine
+from habit_server.db_manager import DBManager
+from habit_server.db_models import User, UserActivity, UserHabit
+from habit_server.__init__ import app, db
+
+class DBManagerTestFixture():
```
Comment (jrdbuch) [inline test/test_api.py:11]: import this from test_db_manager instead of redefining

File: src/habit_server/db_manager.py [hunk 3]
```diff
@@ -15,7 +14,6 @@ def __init__(self, session):
 
     def add_user(self, user):
         """Add a new user to the database.
-
```
Comment (jrdbuch) [inline src/habit_server/db_manager.py:18]: you're gonna run into pydocstyle issues by deleting these blank lines

**Review by jrdbuch — APPROVED** [review]
see above comments, everything else looks good

**Pull request merged**